packages/ui-app/src/Params/Param/Balance.tsx

```diff
@@ -0,0 +1,158 @@
+      info: '',
+      isPreKeyDown: false
+    };
+  }
+
+  render () {
+    const { className, defaultValue: { value }, isError, label, style, t, withLabel } = this.props;
+    const { error, info, warn } = this.state;
+    const defaultValue = new BN((value as BN).toString(10) || '0').toString(10);
+
+    return (
+      <Bare
+        className={className}
+        style={style}
+      >
+        <InputNumber
+          className='large'
+          defaultValue={defaultValue || '0'}
+          error={error}
+          info={info}
+          isError={isError}
+          label={label}
+          maxLength={defaultMaxLength}
+          onChange={this.onChange}
+          onKeyDown={this.onKeyDown}
+          onKeyUp={this.onKeyUp}
+          placeholder={t('account.balance.placeholder', {
+            defaultValue: 'Between 0 DOT and the available DOT balance of the account'
+          })}
+          withLabel={withLabel}
+          warn={warn}
+        />
+      </Bare>
+    );
+  }
+
+  onChange = (value: string): void => {
+    const { onChange, t } = this.props;
+
+    try {
+      const { isValid, errorMessage, infoMessage, warnMessage } = isValidBalance(value, t);
+
+      this.setState({
+        error: !isValid && errorMessage ? errorMessage : '',
+        info: isValid && infoMessage ? infoMessage : '',
+        warn: isValid && warnMessage ? warnMessage : ''
+      });
+
+      if (!onChange) {
+        return;
+      }
+
+      let valueBN: BN;
+      valueBN = new BN(value || '0');
+
+      onChange({
+        isValid,
+        value: valueBN
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  onKeyDown = (event: any): void => {
+    const { t } = this.props;
+    const { isPreKeyDown } = this.state;
+
+    // only allow user balance input to contain one instance of '.' for decimals.
+    // prevent use of shift key
+    if (
+      (keydown.isDuplicateDecimalPoint(event)) ||
+      (keydown.isShift(event))
+    ) {
+      event.preventDefault();
+      return;
+    }
+
+    if (KEYS_ALLOWED.includes(event.keyCode)) {
+      return;
+    }
+
+    if (KEYS_PRE.includes(event.keyCode)) {
+      this.setState({ isPreKeyDown: true });
+    }
+
+    // allow users to to use cut/copy/paste combinations, but not non-numeric letters individually
+    if (
+      (keydown.isSelectAll(event, isPreKeyDown)) ||
+      (keydown.isCut(event, isPreKeyDown)) ||
+      (keydown.isCopy(event, isPreKeyDown)) ||
+      (keydown.isPaste(event, isPreKeyDown))
+    ) {
+      return;
+    }
+
+    // prevent input of non-integer values (allow numeric including from keyboards with numpad)
+    if (keydown.isNonNumeric(event)) {
+      this.setState({
+        error: t('balance.error.format', {
+          defaultValue: 'Balance to transfer in DOTs must be a number'
+        })
+      });
+      event.preventDefault();
+      return;
+    }
+  }
+
+  onKeyUp = (event: any): void => {
+    if (KEYS_PRE.includes(event.keyCode)) {
```

Comment:
Just pass the keyCode in from the underlying component. (Additionally, we now have an untyped `: any`, which we really should not have unless it is an exceptional circumstance.